iphone/Maps/Classes/Search/Filters/MWMSearchHotelsFilterViewController.mm

```diff
@@ -0,0 +1,159 @@
+  [button setAttributedTitle:titleSelected forState:UIControlStateSelected];
+  [button setAttributedTitle:titleSelected
+                    forState:UIControlStateSelected | UIControlStateHighlighted];
+
+  NSDictionary * primaryNormal = @{
+    NSFontAttributeName : regular17,
+    NSForegroundColorAttributeName : [UIColor blackPrimaryText]
+  };
+  NSDictionary * secondaryNormal = @{
+    NSFontAttributeName : regular10,
+    NSForegroundColorAttributeName : [UIColor blackSecondaryText]
+  };
+  NSAttributedString * titleNormal =
+      formString(primaryText, primaryNormal, secondaryText, secondaryNormal);
+  [button setAttributedTitle:titleNormal forState:UIControlStateNormal];
+
+  button.titleLabel.textAlignment = NSTextAlignmentCenter;
+}
+}  // namespace
+
+@interface MWMSearchFilterViewController ()
+
++ (MWMSearchFilterViewController *)controllerWithIdentifier:(NSString *)identifier;
+
+@end
+
+@interface MWMSearchHotelsFilterViewController ()
+
+@property(nonatomic) IBOutletCollection(UIButton) NSArray * ratings;
+
+@property(weak, nonatomic) IBOutlet UIButton * ratingAny;
+@property(weak, nonatomic) IBOutlet UIButton * rating7;
+@property(weak, nonatomic) IBOutlet UIButton * rating8;
+@property(weak, nonatomic) IBOutlet UIButton * rating9;
+
+@property(weak, nonatomic) IBOutlet UIButton * price1;
+@property(weak, nonatomic) IBOutlet UIButton * price2;
+@property(weak, nonatomic) IBOutlet UIButton * price3;
+
+@end
+
+@implementation MWMSearchHotelsFilterViewController
+
++ (MWMSearchHotelsFilterViewController *)controller
+{
+  NSString * identifier = [MWMSearchHotelsFilterViewController className];
+  return static_cast<MWMSearchHotelsFilterViewController *>(
+      [self controllerWithIdentifier:identifier]);
+}
+
+- (void)viewDidLoad
+{
+  [super viewDidLoad];
+  configButton(self.ratingAny, L(@"any"), nil);
+  configButton(self.rating7, L(@"7.0+"), L(@"good"));
+  configButton(self.rating8, L(@"8.0+"), L(@"very_good"));
+  configButton(self.rating9, L(@"9.0+"), L(@"excellent"));
+
+  configButton(self.price1, L(@"$"), nil);
+  configButton(self.price2, L(@"$$"), nil);
+  configButton(self.price3, L(@"$$$"), nil);
+
+  [self changeRating:self.ratingAny];
+
+  self.price1.selected = NO;
+  self.price2.selected = NO;
+  self.price3.selected = NO;
+}
+
+- (shared_ptr<search::hotels_filter::Rule>)getRules
+{
+  using namespace search::hotels_filter;
+  shared_ptr<Rule> ratingRule;
+  if (self.rating7.selected)
+    ratingRule = Ge<Rating>(7.0);
+  else if (self.rating8.selected)
+    ratingRule = Ge<Rating>(8.0);
+  else if (self.rating9.selected)
+    ratingRule = Ge<Rating>(9.0);
+
+  shared_ptr<Rule> priceRule;
+  if (self.price1.selected)
+    priceRule = Or(priceRule, Eq<PriceRate>(1));
+  if (self.price2.selected)
+    priceRule = Or(priceRule, Eq<PriceRate>(2));
+  if (self.price3.selected)
+    priceRule = Or(priceRule, Eq<PriceRate>(3));
+
+  if (!ratingRule && !priceRule)
```

Comment:
I need this check to get nullptr instead of empty rule.
If filter rule is empty, UI must show that there is no rule.
If I return empty rule here, there is no way know whether the rule is empty.